compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -524,6 +524,27 @@ impl DataFlowGraph {
         }
     }
 
+    /// If this value points to an array of constant ASCII bytes, returns a string
+    /// consisting of those bytes.
+    pub(crate) fn get_string(&self, value: ValueId) -> Option<String> {
+        let (value_ids, _typ) = self.get_array_constant(value)?;
+
+        let mut string = String::new();
+        for value_id in value_ids {
+            let field_value = self.get_numeric_constant(value_id)?;
+            let u64_value = field_value.try_to_u64()?;
+            if u64_value > 255 {
+                return None;
+            };
+            let byte = u64_value as u8;
+            if !byte.is_ascii() {
+                return None;
+            }
```

Comment:
Is the ascii restriction necessary?
I wonder if we can't collect everything to a `Vec<u8>` first then use a string conversion method on that instead.

Reply:
This method is also used when printing an array value in SSA, so I guess it could happen that a random byte sequence ends up being valid UTF-8. But maybe not? I'll change it to `String::from_utf8`

Reply:
That said... all our strings are ascii, so maybe this is good enough?

Reply:
I think using `String::from_utf8` should be shorter, still allow ascii, and allow us flexibility in the future when we will eventually want to have more support for utf8 strings

Reply:
Done!